p2p/src/channels/rpc/mod.rs

```diff
@@ -108,15 +109,30 @@ impl Default for P2pRpcRequest {
     }
 }
 
+fn to_binary(bytes: &[u8]) -> String {
+    bytes
+        .into_iter()
+        .copied()
+        .map(|byte| {
+            (0..8)
+                .into_iter()
+                .map(|b| byte & (1 << (7 - b)) != 0)
+                .map(|b| if b { '1' } else { '0' })
+                .collect::<String>()
+        })
+        .collect()
+}
+
 impl std::fmt::Display for P2pRpcRequest {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         write!(f, "{:?}", self.kind())?;
         match self {
             Self::BestTipWithProof => Ok(()),
             Self::LedgerQuery(ledger_hash, query) => {
-                let addr_to_str = |addr| {
-                    let addr = ledger::Address::from(addr);
-                    format!("depth: {}, addr: {}", addr.length(), addr.to_string())
+                let addr_to_str = |addr: &MerkleAddressBinableArgStableV1| {
+                    let depth = &addr.0 .0;
+                    let bits = to_binary(&addr.1[..(*depth as usize)]);
+                    format!("depth: {depth}, addr: {bits}")
                 };
```

Comment:
Here `depth` is the number of bits in `addr.1`. 
```
&addr.1[..(*depth as usize)]
```
This is not correct, because `depth` is not the number of bytes.

That `addr_to_str` should be:
```
fn addr_to_str(addr: &MerkleAddressBinableArgStableV1) -> String {
    let MerkleAddressBinableArgStableV1(depth, bits) = addr;
    let depth = depth.as_u64() as usize;

    let addr: String = (0..depth)
        .map(|index| {
            let byte_index = index / 8;
            let bit_index = index % 8;
            bits[byte_index] & (1 << (7 - bit_index)) != 0
        })
        .map(|b| if b { '1' } else { '0' })
        .collect();

    format!("depth: {depth}, addr: {addr}")
}
```

Reply:
Ah, right, got confused with `depth` and `pos` in the `From` impl. Will fix that.

Reply:
@akoptelov Ah yes, that `pos` should use a different name,